crypto/tss/eddsa/frost/signer/signer_test.go

```diff
@@ -166,3 +199,95 @@ func newSigners(curve elliptic.Curve, expPublic *ecpointgrouplaw.ECPoint, ss [][
 	}
 	return signers, listeners
 }
+
+func Verify(pubKey, R *ecpointgrouplaw.ECPoint, message []byte, s *big.Int) bool {
+	curveType := pubKey.GetCurve()
+	switch curveType {
+	case elliptic.Secp256k1():
+		curveP := curveType.Params().P
+		curveN := curveType.Params().N
+		// Let P = lift_x(int(pk))
+		Px, Py, err := lift_x(pubKey.GetX(), curveType)
+		if err != nil {
+			return false
+		}
+		// Let r = int(sig[0:32]); fail if r ≥ p.
+		r := new(big.Int).Set(R.GetX())
+		if r.Cmp(curveP) >= 0 {
+			return false
+		}
+		// Let s = int(sig[32:64]); fail if s ≥ n.
+		s := new(big.Int).Set(s)
+		if s.Cmp(curveN) >= 0 {
+			return false
+		}
+		// Let e = int(hashBIP0340/challenge(bytes(r) || bytes(P) || m)) mod n.
+		toHash := utils.Bytes32(r)
+		toHash = append(toHash, utils.Bytes32(Px)...)
+		toHash = append(toHash, message...)
+		e := new(big.Int).SetBytes(hash("BIPSchnorr", toHash))
+		e.Mod(e, curveN)
+		// Let R = s⋅G - e⋅P.
+		RecoverPubKey, err := ecpointgrouplaw.NewECPoint(curveType, Px, Py)
+		if err != nil {
+			return false
+		}
+		R1 := ecpointgrouplaw.ScalarBaseMult(curveType, s)
+		R2 := RecoverPubKey.ScalarMult(e)
+		R2 = R2.Neg()
+		compareR, err := R1.Add(R2)
+		if err != nil {
+			return false
+		}
+		// Fail if is_infinite(R).
+		// Fail if not has_even_y(R).
+		// Fail if x(R) ≠ r
+		if compareR.IsIdentity() || !compareR.IsEvenY() || compareR.GetX().Cmp(r) != 0 {
+			return false
+		}
+		return true
+
+	case elliptic.Ed25519():
+		edwardPubKey := edwards.NewPublicKey(edwards.Edwards(), pubKey.GetX(), pubKey.GetY())
+		test1, err := ecpointEncoding(R)
+		Expect(err).Should(BeNil())
+		test2 := *test1
+		r := new(big.Int).SetBytes(utils.ReverseByte(test2[:]))
+		return edwards.Verify(edwardPubKey, message, r, s)
+	}
+	return false
+}
+
+func lift_x(x *big.Int, curve elliptic.Curve) (*big.Int, *big.Int, error) {
```

Comment:
It's recommended to use camelcase. Try `liftX`

Reply:
done